### PR TITLE
Implement KaTeX maths rendering

### PR DESCRIPTION
**Pull request by will-hart**
Re #7. 

This solution is not ideal as the npm repository for `katex` doesn't include CSS files and fonts so these are manually bundled. 

Also, backslashes - `\` - are not properly rendered in the CodeMirror editor on my Win10 install, so maths can look a bit strange in the editor, however maths renders and the markdown file saves and loads as expected.

### CODE REVIEW

**Comment by yhatt** [comment]
Thanks! 👍 
A strange rendering on CodeMirror must be created issue, but I saw there is nothing wrong with rendering on slides.

If ours repository includes katex resources, I just fear that MIT License violation is brought by forgetting inclusion of katex's...

**Comment by will-hart** [comment]
Re the backslash, I get the same behaviour in 0.0.8 binaries downloaded from the release page which don't include KaTeX. I'll create a new issue. 

KaTeX is available under MIT as well - https://github.com/Khan/KaTeX/blob/master/LICENSE.txt

**Comment by yhatt** [comment]
Oops, I could not rendering correctly in packaged binary for production. 💦 
We use `gulp dist` to extract files for production (`gulpfile.coffee:96`), but currently KaTeX's font files is not include.

And, MIT Licenses needs to be clearly showing KaTeX's copyrights. e.g. Like this (with underscore.js MIT): https://github.com/Khan/KaTeX/blob/master/LICENSE.txt

**Comment by will-hart** [comment]
Ah ok, I'll have a look at these tomorrow.

**Comment by will-hart** [comment]
Ok, license file updated and the packaging works (only able to test on Windows)

**Comment by yhatt** [comment]
Thanks!
Hmm, It seems to wrong webfont's path still. The compiled CSS is putting on to `css` directory, it's not `sass`.

Maybe putting on `katex` under `resources` is more better.
I'm sorry to take up your time.

**Comment by will-hart** [comment]
Ok, I've moved the KaTeX files to the resources directory and then added to the gulp `compile:sass` steps so that these files are copied into the CSS folder. 

This means we don't have to modify the KaTeX files with updated paths and it's easy to upgrade KaTeX by just dropping new files on top in the resources directory while these files aren't on npm.

I'm not particularly familiar with `gulp` so let me know if I missed any steps.

File: gulpfile.coffee [hunk 1]
```diff
@@ -103,6 +107,7 @@ gulp.task 'dist', ['clean:dist'], ->
     '*.html'
     'package.json'
     'example.md'
+    'sass/katex/fonts/*'
```
Comment (yhatt) [inline gulpfile.coffee]: Just a little more! This line is no longer using.
I'll merge after you remove this line. I'm grateful for you work! 😆

**Comment by will-hart** [comment]
Oh wait I'm an idiot I think I deleted the wrong line!

**Comment by yhatt** [comment]
Yes, please remove `'sass/katex/fonts/*'` !

**Comment by will-hart** [comment]
Ok hopefully that should fix it. You may need to squash some commits! :smiley:

**Comment by yhatt** [comment]
LGTM!!! :+1:
Thanks to your awesome contribution! I'll merge with `git merge --squash`.

**Comment by yhatt** [comment]
Done! I squashed commits at https://github.com/yhatt/marp/commit/74ada4f19d06fc04d5c64d9430b0d0f87e1a1898 and merged at https://github.com/yhatt/marp/commit/a688d831d8e7183d0ff23f4c45c5b1c9fee25df2

**Pull request closed without merging**